pkg/provider/azure_storageaccount.go

```diff
@@ -354,6 +368,42 @@ func (az *Cloud) EnsureStorageAccount(ctx context.Context, accountOptions *Accou
 		if rerr := az.StorageAccountClient.Create(ctx, subsID, resourceGroup, accountName, cp); rerr != nil {
 			return "", "", fmt.Errorf("failed to create storage account %s, error: %v", accountName, rerr)
 		}
+		defer func() {
+			if retErr != nil {
```

Comment:
not quite sure whether `retErr` would be equal to `fmt.Errorf("failed to set blob service properties for storage account %s, error: %w", accountName, err)`

Reply:
let me do some manual test

Reply:
![image](https://user-images.githubusercontent.com/44308864/212317197-6ed98638-5c47-4652-b584-135bdc15c5dd.png)

![image](https://user-images.githubusercontent.com/44308864/212317828-7108bb53-1bfa-44eb-a884-c266e969ff74.png)

It worked, there was not large amount of storageaccounts in the resource group.